src/drivers/display/fb/lv_linux_fbdev.c

```diff
@@ -312,16 +312,20 @@ static void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * colo
         return;
     }
 
-    uint32_t color_pos = (area->x1 + dsc->vinfo.xoffset) * px_size + area->y1 * dsc->finfo.line_length;
-    uint32_t fb_pos = color_pos + dsc->vinfo.yoffset * dsc->finfo.line_length;
+    uint32_t color_pos =
+        area->x1 * px_size +
+        area->y1 * dsc->vinfo.xres * px_size;
+    uint32_t fb_pos =
+        (area->x1 + dsc->vinfo.xoffset) * px_size +
+        (area->y1 + dsc->vinfo.yoffset) * dsc->finfo.line_length;
 
     uint8_t * fbp = (uint8_t *)dsc->fbp;
     int32_t y;
     if(LV_LINUX_FBDEV_RENDER_MODE == LV_DISPLAY_RENDER_MODE_DIRECT) {
         for(y = area->y1; y <= area->y2; y++) {
             lv_memcpy(&fbp[fb_pos], &color_p[color_pos], w * px_size);
             fb_pos += dsc->finfo.line_length;
-            color_pos += dsc->finfo.line_length;
+            color_pos += dsc->vinfo.xres * px_size;
```

Comment:
I'm coming here from #6002 where I tried to fix the same problem. Interestingly, this change here does not appear to resolve the issue I'm seeing. In my case `dsc->finfo.line_length` is equal to `dsc->vinfo.xres * px_size` so this doesn't make a difference. Using `disp->hor_res` instead of `dsc->vinfo.xres` fixes the rendering for me. @pvginkel would that also work in your case?

Reply:
It's great that the two discussion are joined and hopefully we will have a more robust solution in the end. :+1:

Reply:
@Johennes I must assume that you've rotated the display then. I'm thinking that we replace `xres` with `hres` if rotation is `LV_DISPLAY_ROTATION_90` or `LV_DISPLAY_ROTATION_270`. Can you verify whether this assumption is correct, that the display rotation is set to one of these values?

Reply:
No, I'm actually not using rotation. I'm using `lv_display_set_physical_resolution` and `lv_display_set_resolution` to limit the LVGL display to a subsection of the framebuffer. That's why using the resolution from the framebuffer's `vinfo` breaks things for me.

Reply:
Would you be able to share the code? I must confess I have no idea how to determine the size of the paint buffer in your scenario.

Reply:
Yes, this works for me, too. Thanks a lot. :+1: